apps/block_scout_web/assets/js/lib/smart_contract/wei_ether_converter.js

```
@@ -0,0 +1,23 @@
+import $ from 'jquery'
+import { BigNumber } from 'bignumber.js'
+
+const weiToEtherConverter = (element, event) => {
+  const $element = $(element)
+  const $conversionText = $element.find('[data-conversion-text]')
+  const $conversionUnit = $element.find('[data-conversion-unit]')
+  let unitVal = new BigNumber($conversionUnit.html())
+
+  if (event.target.checked) {
+    $conversionText.html('ETH')
+    unitVal = unitVal / new BigNumber('1000000000000000000')
+    $conversionUnit.html(String(unitVal > 0 ? unitVal : unitVal.toFixed(3)))
+  } else {
+    $conversionText.html('WEI')
```

`WEI`/`ETH` need to be run through gettext.

Since we don't have access to gettext in JS you will need to put both labels in the UI, `style="display: none;"` on the one that is not the default, and then use jQuery `show`/`hide` to display the correct ones when toggling.

---

ok done